### PR TITLE
rename snapshot Engine to RocksEngine; and implements base function.

### PR DESCRIPTION
**Pull request by wenbobuaa**
# Description

1. snapshot.Engine 改名为 RocksEngine ；
2. 实现几个操作 rocksdb 的基本函数；
3. 调整目录结构；

## Type of change

- **New feature**     
- **Refactoring**

# Checklist:

- [x] **Style**:       My code follows the **style guidelines** of this project
- [x] **Self-review**: I have performed a **self-review** of my own code
- [x] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [ ] **Doc**:         I have made corresponding changes to the **documentation**
- [x] **No-warnings**: My changes generate **no new warnings**
- [x] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [x] **Pass**:        New and existing **unit tests pass** locally with my changes
- [ ] **Dep**:         Any **dependent** changes have been merged and published in downstream modules

### CODE REVIEW

File: components/epaxos/src/snapshot/rocks/mod.rs [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
+enum_str! {
+    pub DBColumnFamily {
+        Default("default")
+        Instance("instance")
+        Status("status")
+    }
+}
+
+impl DBColumnFamily {
+    fn all<'a>() -> Vec<&'a str> {
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocks/mod.rs]: 这个函数的测试在这个包里吗?

Comment (wenbobuaa) [reply components/epaxos/src/snapshot/rocks/mod.rs]: 缺测试了。加一个

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    /// ```
+    pub fn new(path: &str) -> Result<RocksDBEngine, Error> {
+        let db = open(path)?;
+
+        Ok(RocksDBEngine { db: db })
+    }
+
+    /// Set a key-value pair to rocksdb.
+    fn _set(&mut self, cf: &DBColumnFamily, k: &[u8], v: &[u8]) -> Result<(), Error> {
+        let cfh = self.db.cf_handle(cf.as_str()).unwrap();
```
Comment (liubaohai) [inline components/epaxos/src/snapshot/rocks/engine.rs]: 如果unwrap会panic，返回一个error好点。unwrap不建议用

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    /// };
+    /// ```
+    pub fn new(path: &str) -> Result<RocksDBEngine, Error> {
+        let db = open(path)?;
+
+        Ok(RocksDBEngine { db: db })
+    }
+
+    /// Set a key-value pair to rocksdb.
+    fn _set(&mut self, cf: &DBColumnFamily, k: &[u8], v: &[u8]) -> Result<(), Error> {
```
Comment (pengsven) [inline components/epaxos/src/snapshot/rocks/engine.rs]: 为啥要_划线开始

Comment (wenbobuaa) [reply components/epaxos/src/snapshot/rocks/engine.rs]: 这些暂时是没用到的函数，会有 warning，所以加了下划线消warning。后面用到之后会改掉。

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 4]
```diff
@@ -26,30 +26,56 @@ impl RocksDBEngine {
...
             Ok(option_val) => match option_val {
                 Some(val) => {
                     return Ok(val.to_vec());
                 }
                 None => {
+                    let k_str = match str::from_utf8(k) {
+                        Ok(s) => s,
+                        Err(err) => {
+                            return Err(Error::RocksDBError {
+                                msg: format!("got error {} while convert utf8 to str", err),
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocks/engine.rs]: got error 去掉吧...感觉在消息里没啥用...
```suggestion
                                msg: format!("{} while converting utf8 to str", err),
```

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 5]
```diff
@@ -26,30 +26,56 @@ impl RocksDBEngine {
...
+                            });
+                        }
+                    };
                     return Err(Error::RocksDBError {
-                        msg: format!("value for key {} not found!", str::from_utf8(k).unwrap()),
+                        msg: format!("value for key {} not found!", k_str),
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocks/engine.rs]: 错误消息里加感叹号还挺奇怪的...
```suggestion
                        msg: format!("key not found: {}", k_str),
```

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 6]
```diff
@@ -26,30 +26,56 @@ impl RocksDBEngine {
...
                     });
                 }
             },
             Err(err) => {
+                let k_str = match str::from_utf8(k) {
+                    Ok(s) => s,
+                    Err(err) => {
+                        return Err(Error::RocksDBError {
+                            msg: format!("got error {} while convert utf8 to str", err),
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocks/engine.rs]: 同上

File: components/epaxos/src/snapshot/rocks/engine.rs [hunk 7]
```diff
@@ -26,30 +26,56 @@ impl RocksDBEngine {
...
+                        });
+                    }
+                };
                 return Err(Error::RocksDBError {
-                    msg: format!("load key {} got error: {}", str::from_utf8(k).unwrap(), err),
+                    msg: format!("load key {} got error: {}", k_str, err),
```
Comment (drmingdrmer) [inline components/epaxos/src/snapshot/rocks/engine.rs]: 统一用xx error while yy的风格?

Comment (wenbobuaa) [reply components/epaxos/src/snapshot/rocks/engine.rs]: 好 我统一一下

**Pull request merged**